Langs/0415.lang

```diff
@@ -36,8 +36,8 @@ Unable to Check List=Nie można sprawdzić listy
 Error Accessing List=Błąd dostępu listy
 Not Currently Listed as Compatible=Obecnie nie na liście kompatybilności
 Listed as Compatible=Na liście kompatybilności
-Cores=Rdzenie
```

Comment:
Pomyślmy tutaj chwilę, dane będą z Steam Hardware Survey https://store.steampowered.com/hwsurvey

Rdzenie używamy przy ilościach: 2, 3, 4. (Też 22,24 itp, ale nie będę ich brał pod uwagę bo są mało popularne)
No to teraz dane użycia:12.82+0.65+38.98=52.45
Czyli w przypadku ponad 50% użytkowników forma rdzenie jest poprawna. Oczywiście najlepszym wyborem byłaby możliwość zmiany  w zależności od liczby, ale jeśli wybór ma być jeden, to chyba lepiej jeśli będzie poprawnie w 52% przypadków niż w 48. Jestem ciekaw co o tym sądzą @BxOxSxS oraz @mlipok .

Reply:
Skoro dla statystycznej większości użytkowników jest dobrze to bym tak zostawił. To samo się tyczy wątków. Steam niestety nie udostępnia o tym danych ale z tego co wiem większość procesów <=4 rdzeni nie ma większej ilości wątków

Reply:
Zamiast

```
Rdzenie 1
Rdzenie 2
Rdzenie 3
Rdzenie 4

```

Użyć należy:
```
Rdzenie: 1
Rdzenie: 2
Rdzenie: 3
Rdzenie: 4

```
i będzie dobrze

Reply:
To jest niemożliwe bo jest zaprogramowane tak aby najpierw była liczba potem wyrażenie np 4 Rdzenie lub 8 Rdzenie. Nie można tego zmienić bez ingerencji w resztę kodu

Reply:
Mała zmiana, myślę, że da się "wynegocjować", wystarczy zmienić kolejność
https://github.com/rcmaehl/WhyNotWin11/blob/4ba87e2fd5092e4566b71b1473b4a38a76b359d9/WhyNotWin11.au3#L569

Reply:
@BxOxSxS  do you refer to this part of code:

```AutoIt
GUICtrlSetData($hCheck[3][2], _GetCPUInfo(0) & " " & _Translate($iMUI, "Cores") & @CRLF & _GetCPUInfo(1) & " " & _Translate($iMUI, "Threads"))
```

??


**To all contributors** Please answer to this foolowing questions:

In which languges there is a need to use:

`Cores #`
instead
`# Cores`

and
`Threads #`
instead
`# Threads`

Reply:
For example
ENGLISH `# Cores` and `# Threads`
POLISH `Cores #` and `Threads #`

Reply:
> To jest niemożliwe bo jest zaprogramowane tak aby najpierw była liczba potem wyrażenie np 4 Rdzenie lub 8 Rdzenie. Nie można tego zmienić bez ingerencji w resztę kodu

It is easy to maintain this differences, but we must to know which `*.lang` files requires which case.

Reply:
> For example
> `0419.lang` ENGLISH US `# Cores` and `# Threads`
> `0415.lang` POLISH `Cores #` and `Threads #`

W angielskim przecież `Cores: 8` też by pasowało

Reply:
> For example
> `0419.lang` ENGLISH US `# Cores` and `# Threads`
> `0415.lang` POLISH `Cores #` and `Threads #`

For this reason I make this requests.


> **To all contributors** Please answer to this foolowing questions:
> In which languges there is a need to use:
> `Cores #`
> instead
> `# Cores`
> 
> and
> `Threads #`
> instead
> `# Threads`

Reply:
Tak o to mi chodzi. Można to zmienić ale to dotyczy wtedy całego projektu i tłumaczeń. Najlepszymi rozwiązaniem by było gdyby liczba rdzeni była zamieniana ze zamiennej z wyrażenia np `Cores=Rdzenie: $i` lub po prostu `Cores=Rdzenie:`

Reply:
I propose to add to Translation:
Cores=# Rdzenie
Cores=Rdzenie #

And parse them by StringReplace()

```AutoIt
Local $sCores = StringReplace(_Translate($iMUI, "Cores"), '#',_GetCPUInfo(0))
If @extended = 0 Then $sCores = _GetCPUInfo(0) & " " $sCores
Local $sThreads = StringReplace(_Translate($iMUI, "Threads"), '#',_GetCPUInfo(1))
If @extended = 0 Then $sThreads = _GetCPUInfo(1) & " " $sThreads
GUICtrlSetData($hCheck[3][2], $sCores & @CRLF & $sThreads)
```

Reply:
@rcmaehl

Reply:
> ```autoit
> Local $sCores = StringReplace(_Translate($iMUI, "Cores"), '#',_GetCPUInfo(0))
> If @extended = 0 Then $sCores = _GetCPUInfo(0) & " " $sCores
> Local $sThreads = StringReplace(_Translate($iMUI, "Threads"), '#',_GetCPUInfo(1))
> If @extended = 0 Then $sThreads = _GetCPUInfo(1) & " " $sThreads
> GUICtrlSetData($hCheck[3][2], $sCores & @CRLF & $sThreads)
> ```

This solution also accepts the current form, ie if there are no "#" characters, it works as before.

Reply:
@kp1971  take a look on yours, PR
https://github.com/kp1971/WhyNotWin11/pulls

Accept/Merge them and this should be visible here.

Reply:
@mlipok As per contributing guidelines "One change per pull request". I think that these 2 should be pull requests into this repo, and if Robert agrees to merge them, then this one can also be merged.

Reply:
It is one REQUEST, I also changed the Langs/0415.lang

@kp1971  please change PR TITLE from:
`Update 0415.lang`
to
`*.lang files - support for # in  Cores/Threads`

Reply:
I had a problem with my PR so I delete this branches, I will do separate PR

Reply:
Take a look here:
https://github.com/rcmaehl/WhyNotWin11/pull/403